uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.controller.js

```diff
@@ -0,0 +1,419 @@
+/* eslint-env amd */
+
+define([
+  'common/angular',
+  'common/moment',
+  'common/lodash',
+  'tasks-assignments/controllers/controllers',
+  'tasks-assignments/services/contact',
+  'tasks-assignments/services/file',
+  'tasks-assignments/services/dialog',
+  'tasks-assignments/services/document',
+  'common/services/notification'
+], function (angular, moment, _, controllers) {
+  'use strict';
+
+  controllers.controller('ModalDocumentCtrl', ModalDocumentCtrl);
+
+  ModalDocumentCtrl.$inject = [
+    '$scope', '$uibModalInstance', '$rootScope', '$rootElement', '$q', '$log', 'role',
+    '$filter', '$uibModal', '$dialog', '$timeout', 'AssignmentService', 'DocumentService',
+    'ContactService', 'FileService', 'data', 'files', 'config', 'HR_settings', 'modalMode',
+    'notification'
+  ];
+
+  function ModalDocumentCtrl ($scope, $modalInstance, $rootScope, $rootElement, $q, $log, role,
+    $filter, $modal, $dialog, $timeout, AssignmentService, DocumentService, ContactService,
+    FileService, data, files, config, HRSettings, modalMode, notification) {
+    $log.debug('Controller: ModalDocumentCtrl');
+
+      // Init call
+    (function init () {
+      $scope.files = [];
```

Comment:
As most of the changes are already done in #239, I prefer to make changes you requested once #239 is merged.